core/src/main/java/org/mule/runtime/core/privileged/routing/CompositeRoutingException.java

```diff
@@ -54,41 +67,83 @@ public CompositeRoutingException(RoutingResult routingResult) {
   @Override
   public String getDetailedMessage() {
     StringBuilder builder = new StringBuilder();
-    builder.append(MESSAGE_TITLE).append(lineSeparator());
-
-    for (Entry<String, Error> entry : routingResult.getFailures().entrySet()) {
-      String routeSubtitle = String.format("Route %s: ", entry.getKey());
-      MuleException muleException = ExceptionHelper.getRootMuleException(entry.getValue().getCause());
-      if (muleException != null) {
-        builder.append(routeSubtitle).append(muleException.getDetailedMessage());
-      } else {
-        builder.append(routeSubtitle)
-            .append("Caught exception in Exception Strategy: " + entry.getValue().getCause().getMessage());
+    Map<String, Pair<Error, EventProcessingException>> detailedFailures = getDetailedFailures();
+
+    if (detailedFailures.isEmpty()) {
```

Comment:
Extract the content (true branch) to a method called getLegacyDetailedMessage() that returns the message String.